internal/runtime/funcs/file_reader.go

```diff
@@ -0,0 +1,57 @@
+package funcs
+
+import (
+	"context"
+	"io"
+	"os"
+
+	"github.com/nevalang/neva/internal/runtime"
+)
+
+type fileReader struct{}
+
+func (c fileReader) Create(rio runtime.FuncIO, msg runtime.Msg) (func(ctx context.Context), error) {
+	filename, err := rio.In.Port("filename")
+	if err != nil {
+		return nil, err
+	}
+
+	dataPort, err := rio.Out.Port("data")
+	if err != nil {
+		return nil, err
+	}
+
+	errPort, err := rio.Out.Port("err")
+	if err != nil {
+		return nil, err
+	}
+
+	return func(ctx context.Context) {
+		select {
+		case <-ctx.Done():
+			return
+		case name := <-filename:
```

Comment:
your implementation will only handle one message, every runtime function should have for loop to handle infinite number of messages, checkout other runtime functions for inspiration